woocommerce/google-listings-and-ads
# Split shipping rates and shipping times section, and add automatic shipping rates option

### Changes proposed in this Pull Request:

Uses shipping rates API changes introduced in PR #1192.

Notable UI changes:

- Split the `CombinedShipping` UI into separate "Shipping rates" and "Shipping times" sections.
- Add "automatic" shipping rates option, and have it set as initial default value in Setup MC.

Other technical changes:

- Add "automatic" as a valid shipping rate value in `checkErrors` function.
- Add and fix tests for automatic shipping rate.
- Remove `CombinedShipping` from the code base.
    - Related `gla-shipping-title` CSS class is also removed and no longer in use in the code base.

Note: Changes to the shipping rate setup (i.e. minimum order amount for free shipping) will be done in a separate PR.

### References:

Figma: 2b5f7-pb

Internal P2 post: pcTzPl-ep-p2

### Screenshots:

Setup MC flow:

![image](https://user-images.githubusercontent.com/417342/149899389-679950e4-2603-4c80-995c-d4f25b892196.png)

Edit free listings:

![image](https://user-images.githubusercontent.com/417342/149899422-de424060-029f-4285-9c05-8d5e66ac25f3.png)

📹  Demo video with my voice annotation showing the Setup MC flow:

https://user-images.githubusercontent.com/417342/149969997-5fcb00bb-5c42-4173-8eba-5f29c0f6d374.mp4

### Detailed test instructions:


1. Go through Setup MC flow.
2. In Step 3 Configure your product listings: 
    - You should see separate shipping rates section and shipping times section.
    - The "automatic" shipping rate option should be selected by default. 
    - Shipping rate detailed setup should only be displayed when you select the "simple" option in shipping rate section.
    - Shipping time detailed setup should only be displayed when you select the "simple" option in shipping time section.
3. Complete the Setup MC flow. 
4. Go to the plugin's dashboard page, and edit the free listings program.
5. Go to Configure your product listings and edit the values there. It should behave similar to the Setup MC flow above.



### Changelog entry

> Add - Split shipping into separate shipping rates section and shipping times section, and add automatic option to shipping rates
